### PR TITLE
Core/Movement Handle Movement Forces

### PR DESCRIPTION
**Pull request by Traesh**
**Changes proposed:**

- Add a system to add and remove movement forces from unit
- Will allow to script spells using this system (in panda/wod/legion dungeons mainly)

**Target branch(es):** master
- [x] master

**Issues addressed:**
None

**Tests performed:**
None on this TC update, lot on Ashamane using the same code

**Known issues and TODO list:**

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -13632,6 +13632,69 @@ void Unit::SendSetVehicleRecId(uint32 vehicleId)
...
+    moveRemoveMovementForce.MoverGUID = GetGUID();
+    moveRemoveMovementForce.SequenceIndex = m_movementCounter++;
+    moveRemoveMovementForce.ID = source;
+    SendMessageToSet(moveRemoveMovementForce.Write(), true);
+}
+
+void Unit::RemoveAllMovementForces()
+{
+    // We need to copy the map because RemoveMovementForce method will delete from original map
+    std::unordered_map<ObjectGuid, WorldPackets::Movement::MovementForce> movementForcesCopy = _movementForces;
```
Comment (Warpten) [inline src/server/game/Entities/Unit/Unit.cpp]: while (!empty()) { delete front }

**Comment by Shauren** [comment]
Note to self when fixing it up for merge: opcodes are all wrong, need to send _UPDATE to nearby and the one with SequenceIndex field is self-only

**Comment by Seyden** [comment]
Btw is movement force really only clientside? doesnt make kind of sense to me if you could block it clientside to prevent a lot of mechanics

**Comment by Shauren** [comment]
@Seyden do you currently see any checks on regular movement in TC? No, you don't (almost, except for cases than can crash/freeze other players), anticheat patches do that

**Comment by Shauren** [comment]
Fixing it turned out to be more than a few lines here and there

**Pull request closed without merging**